testsuite/tests/typing-modules/inclusion_errors.ml

```diff
@@ -1853,3 +1853,183 @@ Error: Signature mismatch:
+         type t = x:int -> int
+       is not included in
+         type t = int -> int
+       The type "x:int -> int" is not equal to the type "int -> int"
+       The argument "x" is labeled, but an unlabeled argument was expected
+|}]
+
+module Eq_label2: sig
+  type t = y:int -> int
+end = struct
+  type t = x:int -> int
+end
+[%%expect {|
+Lines 3-5, characters 6-3:
+3 | ......struct
+4 |   type t = x:int -> int
+5 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig type t = x:int -> int end
+       is not included in
+         sig type t = y:int -> int end
+       Type declarations do not match:
+         type t = x:int -> int
+       is not included in
+         type t = y:int -> int
+       The type "x:int -> int" is not equal to the type "y:int -> int"
+       Labels "x" and "y" do not match
+|}]
+
+module Label1 : sig
+  val f: int -> unit
+end = struct
+  let f ~x = ()
+end
+[%%expect {|
+Lines 3-5, characters 6-3:
+3 | ......struct
+4 |   let f ~x = ()
+5 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig val f : x:'a -> unit end
+       is not included in
+         sig val f : int -> unit end
+       Values do not match:
+         val f : x:'a -> unit
+       is not included in
+         val f : int -> unit
+       The type "x:'a -> unit" is not compatible with the type "int -> unit"
+       The argument "x" is labeled, but an unlabeled argument was expected
+|}]
+
+module Label2 : sig
+  val f: int -> unit
+end = struct
+  let f ?x = ()
+end
+[%%expect {|
+Line 4, characters 9-10:
+4 |   let f ?x = ()
+             ^
+Warning 16 [unerasable-optional-argument]: this optional argument cannot be erased.
+
+Lines 3-5, characters 6-3:
+3 | ......struct
+4 |   let f ?x = ()
+5 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig val f : ?x:'a -> unit end
+       is not included in
+         sig val f : int -> unit end
+       Values do not match:
+         val f : ?x:'a -> unit
+       is not included in
+         val f : int -> unit
+       The type "?x:'a -> unit" is not compatible with the type "int -> unit"
+       The argument "?x" is optional, but an unlabeled argument was expected
+|}]
+
+
+module Label3 : sig
+  val f: x:int -> unit
+end = struct
+  let f ?x = ()
+end
+[%%expect {|
+Line 4, characters 9-10:
+4 |   let f ?x = ()
+             ^
+Warning 16 [unerasable-optional-argument]: this optional argument cannot be erased.
+
+Lines 3-5, characters 6-3:
+3 | ......struct
+4 |   let f ?x = ()
+5 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig val f : ?x:'a -> unit end
+       is not included in
+         sig val f : x:int -> unit end
+       Values do not match:
+         val f : ?x:'a -> unit
+       is not included in
+         val f : x:int -> unit
+       The type "?x:'a -> unit" is not compatible with the type "x:int -> unit"
+       The optional argument "?x" was not expected to be optional
```

Comment:
`Labels "?x" and "x" do not match.`

Reply:
I disagree with repeating twice the same name with one minute change.
I propose: `Label "?x" was expected to not be optional`.

Reply:
Wait, what would the error be on a mismatch between `?x:int -> unit` and `y:int -> unit`? I assumed that the use of `x` on both sides was irrelevant to the current situation and wording.

Reply:
In this case, `Labels "?x" and "y" do not match` seems a good simplification to me (the previous error message was `The optional argument %a was expected to be a labeled argument named %a`)